src/zeroconf/_services/info.py

```diff
@@ -180,6 +184,10 @@ def __init__(
         self.other_ttl = other_ttl
         self.interface_index = interface_index
         self._new_records_futures: Set[asyncio.Future] = set()
+        self._dns_address_cache: Optional[List[DNSAddress]] = None
+        self._dns_pointer_cache: Optional[DNSPointer] = None
+        self._dns_service_cache: Optional[DNSService] = None
+        self._dns_text_cache: Optional[DNSText] = None
```

Comment:
Ideally we would have required a subclass of ServiceInfo to be registered with the registry but thats a design choice we would have needed to make many years ago and would be a breaking change now